packages/rocketchat-search/server/service/validationService.js

```diff
@@ -0,0 +1,51 @@
+import _ from 'underscore';
+
+import SearchLogger from '../logger/logger';
+
+class ValidationService {
+	constructor() {}
+
+	validateSearchResult(result) {
+
+		const uid = Meteor.userId();
+		//get subscription for message
+		if (result.message) {
+			result.message.docs = _.chain(result.message.docs)
+				.each((msg) => {
+					const subscription = Meteor.call('canAccessRoom', msg.rid, uid);
```

Comment:
`canAccessRoom` method has many db calls and can be a bit heavy on server. I would at least cache results from same `roomId` and `userId` instead of calling it for each message.